lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts

```diff
@@ -57,164 +82,22 @@ describe('AttachFileCloudWidgetComponent', () => {
-                name: 'staticValue',
-                value: '-root-',
-                type: 'value'
-            }
-        }
-    };
-
-    const allSourceWithWrongAliasParams = {
-        fileSource: {
-            name: 'all file sources',
-            serviceId: 'all-file-sources',
-            destinationFolderPath: {
-                name: 'staticValue',
-                value: '-wrongAlias-',
-                type: 'value'
-            }
-        }
-    };
-
-    const allSourceWithNoAliasParams = {
-        fileSource: {
-            name: 'all file sources',
-            serviceId: 'all-file-sources',
-            destinationFolderPath: {
-                name: 'staticValue',
-                value: '/noalias/createdFolder',
-                type: 'value'
-            }
-        }
-    };
-
-    const allSourceWithoutDestinationFolderPath = {
-        fileSource: {
-            name: 'all file sources',
-            serviceId: 'all-file-sources'
-        }
-    };
-
-    const allSourceWithoutValueProperty = {
-        fileSource: {
-            name: 'all file sources',
-            serviceId: 'all-file-sources',
-            destinationFolderPath: '-mockAlias-'
-        }
-    };
-
-    const fakeMinimalNode: Node = <Node> {
-        id: 'fake',
-        name: 'fake-name',
-        content: {
-            mimeType: 'application/pdf'
-        }
-    };
-
-    const fakeNodeWithProperties: Node = <Node> {
-        id: 'fake-properties',
-        name: 'fake-properties-name',
-        content: {
-            mimeType: 'application/pdf'
-        },
-        properties: {
-            'pfx:property_one': 'testValue',
-            'pfx:property_two': true
-        }
-    };
-
-    const expectedValues = { pfx_property_one: 'testValue', pfx_property_two: true };
-
-    const mockNodeId = new Promise(function (resolve) {
-        resolve('mock-node-id');
-    });
+    function createUploadWidgetField(form: FormModel, fieldId: string, value?: any, params?: any, multiple?: boolean, name?: string, readOnly?: boolean) {
+        widget.field = new FormFieldModel(form, {
+            type: FormFieldTypes.UPLOAD,
+            value: value,
+            id: fieldId,
+            readOnly: readOnly,
+            name: name,
+            tooltip: 'attach file widget',
+            params: <FormFieldMetadata> { ...params, multiple: multiple }
+        });
+    }
 
-    const fakeLocalPngAnswer = {
-        id: 1155,
-        nodeId: 1155,
-        name: 'a_png_file.png',
-        created: '2017-07-25T17:17:37.099Z',
-        createdBy: {
-            id: 1001,
-            firstName: 'Admin',
-            lastName: 'admin',
-            email: 'admin'
-        },
-        relatedContent: false,
-        contentAvailable: true,
-        link: false,
-        mimeType: 'image/png',
-        simpleType: 'image',
-        previewStatus: 'queued',
-        thumbnailStatus: 'queued'
-    };
+    function clickOnAttachButton(id: string) {
```

Comment:
is this clicking the attach button or the attach widget?

Reply:
Updated, Thanks